### PR TITLE
Added support for Dystopia keyboard

### PR DESCRIPTION
**Pull request by Sanctified-git**
## Description

Added all the necessary files to support the keyboard in the keyboards/sanctified/dystopia directory, Added a default and a via keymap. Tested that everything works on a prototype board and ran linting to make sure no glaring issues would be pushed

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/sanctified/dystopia/config.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x514d // "QM"
```
Comment (drashna) [inline keyboards/sanctified/dystopia/config.h:22]:
```suggestion
#define VENDOR_ID       0x514D // "QM"
```

File: keyboards/sanctified/dystopia/dystopia.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B, K2C, K2D, \
+    K30, K31, K32, K33, K34, K35, K36, K37, K38, K39, K3A, K3B, K3C, K3D, \
+    K40, K41, K42,      K44, K45, K46, K47, K48, K49, K4A, K4B, K4C, K4D \
+) { \
+    { K00, K01, K02, K03, K04, K05, K06, K07, K08, K09, K0A, K0B, K0C, K0D }, \
+    { K10, K11, K12, K13, K14, K15, K16, K17, K18, K19, K1A, K1B, K1C, K1D }, \
+    { K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B, K2C, K2D }, \
+    { K30, K31, K32, K33, K34, K35, K36, K37, K38, K39, K3A, K3B, K3C, K3D }, \
+    { K40, K41, K42, KNO, K44, K45, K46, K47, K48, K49, K4A, K4B, K4C, K4D } \
+}
```
Comment (drashna) [inline keyboards/sanctified/dystopia/dystopia.h:36]:
```suggestion
#define ___ KC_NO

#define LAYOUT_split_bs( \
    K00, K01, K02, K03, K04, K05, K06, K07, K08, K09, K0A, K0B, K0C, K0D, \
    K10, K11, K12, K13, K14, K15, K16, K17, K18, K19, K1A, K1B, K1C, K1D, \
    K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B, K2C, K2D, \
    K30, K31, K32, K33, K34, K35, K36, K37, K38, K39, K3A, K3B, K3C, K3D, \
    K40, K41, K42,      K44, K45, K46, K47, K48, K49, K4A, K4B, K4C, K4D \
) { \
    { K00, K01, K02, K03, K04, K05, K06, K07, K08, K09, K0A, K0B, K0C, K0D }, \
    { K10, K11, K12, K13, K14, K15, K16, K17, K18, K19, K1A, K1B, K1C, K1D }, \
    { K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B, K2C, K2D }, \
    { K30, K31, K32, K33, K34, K35, K36, K37, K38, K39, K3A, K3B, K3C, K3D }, \
    { K40, K41, K42, ___, K44, K45, K46, K47, K48, K49, K4A, K4B, K4C, K4D } \
}
```

File: keyboards/sanctified/dystopia/info.json [hunk 3]
```diff
@@ -0,0 +1,84 @@
+{
+    "keyboard_name": "Dystopia",
+    "url": "",
+    "maintainer": "Sanctified",
+    "width": 17.62,
+    "height": 5.716792398988211,
+    "layouts": {
+        "LAYOUT": {
```
Comment (drashna) [inline keyboards/sanctified/dystopia/info.json:8]: This needs to match the layout macro: 
```suggestion
        "LAYOUT_split_bs": {
```

File: keyboards/sanctified/dystopia/readme.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Dystopia
```
Comment (drashna) [inline keyboards/sanctified/dystopia/readme.md:1]: Could you add instructions on how to enter the bootloader, to reflash the keyboard?

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/sanctified/dystopia/readme.md (hunk 4 above)
Comment (Sanctified-git) [reply keyboards/sanctified/dystopia/readme.md:1]: Done in commit 05cf339

**Comment by Sanctified-git** [comment]
Thanks a lot for your review @drashna it's very helpful as it's my first time submitting a PR for QMK , I've committed all the changes you requested,

**Comment by drashna** [comment]
You are very welcome!

**Review by drashna — APPROVED** [review]

File: keyboards/sanctified/dystopia/rules.mk [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+CONSOLE_ENABLE = no        	# Console for debug
+COMMAND_ENABLE = no        	# Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
```
Comment (fauxpark) [inline keyboards/sanctified/dystopia/rules.mk:23]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no       	# Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no        	# Console for debug
COMMAND_ENABLE = no        	# Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
LTO_ENABLE = yes            # Enable link time optimization
```

File: keyboards/sanctified/dystopia/readme.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+![Dystopia](https://imgur.com/cwZ8bVV.jpg)
```
Comment (fauxpark) [inline keyboards/sanctified/dystopia/readme.md:3]:
```suggestion
![Dystopia](https://imgur.com/cwZ8bVVl.jpg)
```

File: keyboards/sanctified/dystopia/readme.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+Ergonomic 65% layout keyboard 
+Features an 8 degree alpha rotation (similar to Prime-E and Meridian)
```
Comment (fauxpark) [inline keyboards/sanctified/dystopia/readme.md:6]:
```suggestion
Ergonomic 65% layout keyboard. Features an 8 degree alpha rotation (similar to Prime-E and Meridian).
```

File: keyboards/sanctified/dystopia/readme.md [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+
+* Keyboard Maintainer : [Sanctified](https://github.com/Sanctified-git)
+* Made for the Dystopia PCB & Case
+
+Make example for this keyboard (after setting up your build environment):
+    
+    make sanctified/dystopia:default
+
+To enter bootloader mode, hold the RESET switch on the back of the PCB while plugging it in via USB
```
Comment (fauxpark) [inline keyboards/sanctified/dystopia/readme.md:15]:
```suggestion
To enter bootloader mode, hold the RESET switch on the back of the PCB while plugging it in via USB.
```

File: keyboards/sanctified/dystopia/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+        KC_TRNS,    KC_TRNS,    KC_TRNS,                KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS
+        ),
+    [4] = LAYOUT_split_bs(
+        KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,
+        KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,
+        KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,
+        KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,
+        KC_TRNS,    KC_TRNS,    KC_TRNS,                KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS
+        ),
```
Comment (fauxpark) [inline keyboards/sanctified/dystopia/keymaps/default/keymap.c:47]:
```suggestion
```

File: keyboards/sanctified/dystopia/info.json [hunk 10]
```diff
@@ -0,0 +1,83 @@
+{
```
Comment (fauxpark) [inline keyboards/sanctified/dystopia/info.json:1]: Please redo this file without the rotation, it causes the Configurator to render this instead:
image

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/sanctified/dystopia/info.json (hunk 10 above)
Comment (Sanctified-git) [reply keyboards/sanctified/dystopia/info.json:1]: Thank you for pointing this out, I've updated the file for it to display better with commit 17a8ca3

**Review by fauxpark — CHANGES_REQUESTED** [review]
Looks good but there are some overlapping keys on Caps Lock and Backspace:
image

**Comment by Sanctified-git** [comment]
I've removed the overlapping keys for Backspace as well as Caps Lock

File: keyboards/sanctified/dystopia/config.h [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x514D // "QM"
```
Comment (Erovia) [inline keyboards/sanctified/dystopia/config.h:22]:
```suggestion
#define VENDOR_ID       0x5357 // "SW"
```
I think this would make more sense.

Comment (Sanctified-git) [reply keyboards/sanctified/dystopia/config.h:22]: That's true, I had put my initials but it makes more sense that way

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**